thing/index.js

```diff
@@ -515,7 +526,20 @@ function ThingShadowsClient(deviceOptions, thingShadowOptions) {
          //
          if (thingShadows[thingName].persistentSubscribe === true) {
             this._handleSubscriptions(thingName, ['update', 'get', 'delete'], ['accepted', 'rejected'],
-               'subscribe');
+               'subscribe', function () {
```

Comment:
subscribe callback should check granted subscriptions, see the underneath API here - https://github.com/mqttjs/MQTT.js#subscribe

If not all the topic subscriptions are granted, we could return an error in the register callback, and potentially fail following shadow operations.